auth-api/src/auth_api/services/affiliation_invitation.py

```diff
@@ -0,0 +1,391 @@
+from itsdangerous import URLSafeTimedSerializer
+from jinja2 import Environment, FileSystemLoader
+from requests.exceptions import HTTPError
+from sbc_common_components.tracing.service_tracing import ServiceTracing  # noqa: I001
+
+from auth_api.config import get_named_config
+from auth_api.exceptions import BusinessException, ServiceUnavailableException
+from auth_api.exceptions.errors import Error
+from auth_api.models import AccountLoginOptions as AccountLoginOptionsModel
+from auth_api.models import AffiliationInvitation as AffiliationInvitationModel
+from auth_api.models import InvitationStatus as InvitationStatusModel
+from auth_api.models import Membership as MembershipModel
+from auth_api.models.affiliation import Affiliation as AffiliationModel
+from auth_api.models.entity import Entity as EntityModel
+from auth_api.models.org import Org as OrgModel
+from auth_api.schemas import AffiliationInvitationSchema
+from auth_api.services.entity import Entity as EntityService
+from auth_api.services.user import User as UserService
+from auth_api.utils.enums import AccessType, InvitationStatus, LoginSource, Status
+from auth_api.utils.roles import ADMIN, COORDINATOR, STAFF
+from auth_api.utils.user_context import UserContext, user_context
+
+from ..utils.account_mailer import publish_to_mailer
+from ..utils.util import escape_wam_friendly_url
+from .authorization import check_auth
+from .rest_service import RestService
+
+
+ENV = Environment(loader=FileSystemLoader('.'), autoescape=True)
+CONFIG = get_named_config()
+
+
+class AffiliationInvitation:
+    """Manages Affiliation Invitation data.
+
+    This service manages creating, updating, and retrieving Affiliation Invitation data via
+    the Affiliation Invitation model.
+    """
+
+    def __init__(self, model):
+        """Return an affiliation invitation service instance."""
+        self._model = model
+
+    @ServiceTracing.disable_tracing
+    def as_dict(self):
+        """Return the Affiliation Invitation model as a dictionary."""
+        affiliation_invitation_schema = AffiliationInvitationSchema()
+        obj = affiliation_invitation_schema.dump(self._model, many=False)
+        return obj
+
+    @staticmethod
+    @user_context
+    def create_affiliation_invitation(affiliation_invitation_info: Dict,
+                                      # pylint:disable=unused-argument,too-many-locals
+                                      user, invitation_origin, **kwargs):
+        """Create a new affiliation invitation."""
+        context_path = CONFIG.AUTH_WEB_TOKEN_CONFIRM_PATH
+        from_org_id = affiliation_invitation_info['fromOrgId']
+        to_org_id = affiliation_invitation_info['toOrgId']
+        business_identifier = affiliation_invitation_info['businessIdentifier']
+
+        check_auth(org_id=from_org_id, one_of_roles=(ADMIN, COORDINATOR, STAFF))
+
+        # Validate from/to organizations exists
+        from_org: OrgModel = OrgModel.find_by_org_id(from_org_id)
```

Comment:
`if not from_org := OrgModel.find_by_org_id(from_org_id): ` I think could work for most of these

Reply:
Cleaned this up and some of the subsequent checks to do this